### DISCUSSION TITLE
Bulk insert thousands of records results in ER_PS_MANY_PARAM. Should I split my records?

### DISCUSSION THREAD

**Discussion by zoell** [sidorares/node-mysql2#2722] [Q&A]
I am trying to insert 50K records using the example here:
https://github.com/sidorares/node-mysql2/issues/2168#issuecomment-1672529205

But I am getting an error like this:
code: 'ER_PS_MANY_PARAM',
errno: 1390,
sqlState: 'HY000',
sqlMessage: 'Prepared statement contains too many placeholders',

Looks like this is a MySQL limitation and the limit is 65536. I have 14 placeholders so I can insert 4681 at once. What is the best way to insert more? Just split it and call connection.execute as many times I need?

**Comment by sidorares** [comment]
Split it into much smaller batches.
Alternatively use SELECT INFILE and pass readable stream with your data csv
